### PR TITLE
[Metricbeat][Kubernetes] Add support for 1.26 K8s version

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?
Adds support for Kubernetes 1.26.x version and removes support for 1.22.x

## Related issues

- Relates to https://github.com/elastic/observability-dev/issues/2345

### CODE REVIEW

**Comment by constanca-m** [comment]
> @constanca-m should we also update the multiple metrics.1.25 test files with the latest 1.26 version?

That's a good idea @gsantoro. I will do that.

**Comment by tetianakravchenko** [comment]
> @constanca-m should we also update the multiple metrics.1.25 test files with the latest 1.26 version?

maybe not update, but add 1.26 test files, to make sure that we support both version

File: metricbeat/module/kubernetes/scheduler/_meta/data.json [hunk 1]
```diff
@@ -1,47 +1,38 @@
 {
-    "@timestamp": "2022-12-30T17:23:42.218Z",
-    "ecs": {
-        "version": "8.0.0"
-    },
-    "host": {
-        "name": "fournodecluster-control-plane"
```
Comment (gsantoro) [inline metricbeat/module/kubernetes/scheduler/_meta/data.json:7]: where are these changes coming from?

**Review by gizas — APPROVED** [review]

**Comment by gizas** [comment]
> > @constanca-m should we also update the multiple metrics.1.25 test files with the latest 1.26 version?
> 
> maybe not update, but add 1.26 test files, to make sure that we support both version

Can you also update tests for the rest of datasets? 

(https://github.com/elastic/observability-dev/issues/2345#issuecomment-1375511090)

**Comment by constanca-m** [comment]
> > > @constanca-m should we also update the multiple metrics.1.25 test files with the latest 1.26 version?
> > 
> > 
> > maybe not update, but add 1.26 test files, to make sure that we support both version
> 
> Can you also update tests for the rest of datasets?
> 
> ([elastic/observability-dev#2345 (comment)](https://github.com/elastic/observability-dev/issues/2345#issuecomment-1375511090))

I don't the others use metrics specific to a version, only those 3

File: metricbeat/module/kubernetes/controllermanager/_meta/data.json [hunk 2]
```diff
@@ -1,58 +1,35 @@
-
```
Comment (tetianakravchenko) [inline metricbeat/module/kubernetes/controllermanager/_meta/data.json:1]: are changes in this file relevant to this PR?

File: metricbeat/module/kubernetes/kubernetes.yml [hunk 3]
```diff
@@ -43,7 +43,7 @@ spec:
     spec:
       containers:
       - name: kube-state-metrics
-        image: registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0
+        image: registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.7.0
```
Comment (tetianakravchenko) [inline metricbeat/module/kubernetes/kubernetes.yml:46]: it also would need test files updates for `state-*` metricsets, maybe move this in another PR?

**Review by tetianakravchenko — COMMENTED** [review]
shouldn't `apiserver` metricset tests should be updated as well? not sure why there is used version `2.0` though in file name

File: metricbeat/module/kubernetes/kubernetes.yml (hunk 3 above)
Comment (gizas) [reply metricbeat/module/kubernetes/kubernetes.yml:46]: This is the tests I was referring as well Constanca here eg.https://github.com/elastic/beats/tree/main/metricbeat/module/kubernetes/state_container/_meta/test

I agree this can be done on a new PR

**Comment by tetianakravchenko** [comment]
@constanca-m did you have a chance to check regarding apiserver tests?

> shouldn't `apiserver` metricset tests should be updated as well? not sure why there is used version 2.0 though in file name

File: metricbeat/module/kubernetes/kubernetes.yml (hunk 3 above)
Comment (tetianakravchenko) [reply metricbeat/module/kubernetes/kubernetes.yml:46]: we should update the template issue for updating k8s version and add this step as well there

**Comment by constanca-m** [comment]
> @constanca-m did you have a chance to check regarding apiserver tests?
> 
> > shouldn't `apiserver` metricset tests should be updated as well? not sure why there is used version 2.0 though in file name

Not yet @tetianakravchenko. I'm checking kube state metrics, but I will take a look to that as well
